### PR TITLE
Link field for the full-article-image.

### PR DESCRIPTION
**Pull request by bonzani**
### Summary of Changes
This PR adds a modal link field that can be used to specify an url along with the title and rel attributes and uses it to provide an optional link for the full article image. This is very useful if a lightbox for images is used.
### Testing Instructions
1. Open the edit view of some article with a full article image in the backend.
2. Click on select next to the new field "Full image link"
3. Insert some values in the modal window.
4. Click save and also save the article.
5. Check if the article image in the frontend is now surrounded by the specified link with the according attributes.
### Documentation Changes Required
Add the new link field to the documentation.

### CODE REVIEW

File: layouts/joomla/form/field/link.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+JHtml::_('jquery.framework');
+
+// Initialize some field attributes.
+$attr = !empty($class) ? ' class="' . $class . '"' : '';
+$attr .= !empty($size) ? ' size="' . $size . '"' : '';
+
+// The hidden text field.
+$jsonvalue = $value ? htmlspecialchars($value, ENT_COMPAT, 'UTF-8') : '';
+
+echo '<div class="input-prepend input-append">'
```
Comment (yvesh) [inline layouts/joomla/form/field/link.php]: Can you maybe switch to html output already here?

```
?>
<div class="...">
        <input type="hidden" name="<?php echo $name; ?>" 
```

...
Thank you!

File: libraries/cms/form/field/link.php [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+ * @license     GNU General Public License version 2 or later; see LICENSE
+ */
+
+defined('JPATH_PLATFORM') or die;
+
+/**
+ * Form Field class for the Joomla CMS.
+ * Provides a modal link editor
+ *
+ * @since  3.7
```
Comment (yvesh) [inline libraries/cms/form/field/link.php]: Please use `__DEPLOY_VERSION__` everywhere

File: administrator/components/com_content/views/link/tmpl/default.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+		 *
+		 * @return void
+		 */
+		close: function()
+		{
+			window.parent.jQuery('.modal.in').modal('hide');
+			window.parent.jModalClose();
+		}
+	};
+}(jQuery));
```
Comment (yvesh) [inline administrator/components/com_content/views/link/tmpl/default.php]: Any way to move that jQuery code to an external file? (Testability)

File: administrator/components/com_content/views/link/view.html.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ * @copyright   Copyright (C) 2005 - 2016 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die;
+
+/**
+ * HTML View class for the Link component
+ *
+ * @since  3.7
```
Comment (yvesh) [inline administrator/components/com_content/views/link/view.html.php]: Please use `__DEPLOY_VERSION__` everywhere

File: administrator/components/com_content/views/link/tmpl/default.php (hunk 3 above)
Comment (bonzani) [reply administrator/components/com_content/views/link/tmpl/default.php]: Yes for sure. Any preferences where to put it?

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on ffa43131b2da17abc651bcc4c9609356f20b0cde<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13260">issues.joomla.org/tracker/joomla-cms/13260</a>.</sub>

**Comment by genesisfan** [comment]
I have tested this item :white_check_mark: successfully on ffa43131b2da17abc651bcc4c9609356f20b0cde<br /><br />
I preferred "Full Image Link" instead of "Full image link" as title and as description.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13260">issues.joomla.org/tracker/joomla-cms/13260</a>.</sub>

**Comment by jeckodevelopment** [comment]
@bonzani can you please look at the conflicts?

**Comment by Bakual** [comment]
While this works, I don't think it's such a great idea for core. It sounds more like something which could be achieved with the new custom fields feature and a layout override.
Beside that, there are some UX issues with it.
There is no indication in the field that there is something set. You need to open the modal for it.
The modal itself is actually not needed, better do all in the article form. Using showon, you can show the  additional fields if needed. So I would ditch the modal.
The new fields in the modal also would need some tooltips/descriptions.

**Comment by korneliusz401** [comment]
I have tested this item successfully ✅.

It is a good idea to make images clickable.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13260">issues.joomla.org/tracker/joomla-cms/13260</a>.</sub>

**Comment by HLeithner** [comment]
Hi @bonzani,

this PR will not be merged in to J3.x because of the feature freeze we have, I'm also not confident how this "problem" is solved. For me it looks a bit overengineered for a simple link.

Anyway if you like to add this to core you should create a RFC issue and maybe rebase your PR for J4.

thx for you work and time to improve Joomla!

**Pull request closed without merging**